src/content/development/contribution-guide/_index.en.md

```diff
@@ -234,6 +235,16 @@ During the week when the general availability release is performed, the next ver
 Additionally, a retrospective meeting for the last release cycle will be held.
 There will be no dedicated [test day](#test-days), as the release candidate has been tested and no changes have occurred since.
 
+##### Release Notes
+
+[Release notes] should be published for each GA version.
+All new features and bug fixes of the release should be labeled with `release-note-needed` and have a relevant release note
+documented inside the issue / PR.
```

Comment:
So exactly how is this going to work, technically?

To me it makes most sense to keep as files inside some directory, e.g. `release-notes` or `doc/release-notes`.
This way they won't conflict with each other when cherry picking.
This option lends itself to automation quite nicely.

Another option is on the PR itself *but* if the PR gets reverted later it would be hard to identify which could leave to mismatching notes. Granted, it's a low possibility of this happening, but not zero.
This option would be rather harder to automate, but possible.

Reply:
> To me it makes most sense to keep as files inside some directory, e.g. `release-notes` or `doc/release-notes`. This way they won't conflict with each other when cherry picking. This option lends itself to automation quite nicely.

+1 to using a separate files. I think this is what openstack uses 
https://github.com/openstack/neutron/tree/master/releasenotes/notes
Sample PR: https://github.com/openstack/neutron/commit/c16f88d1481f59e8547c406ac24b670723a3b1da

Reply:
I'm fine with a `release-notes` directory. That would help with multi-branches and backports/reverts.

Another option is just to send release notes directly to the website, not have an intermediary step of writing them in the PR. If the `release-note-needed` label is present, automation could enforce that a commit message includes a link to a website PR.